Tech-Workers-Coalition-Italia/mobilizon-reshare
# Introduced FastAPI and Postgres support

- storage: publications_with_status: Return a list[Publication].
- cli: cli: Add --version.
- cli: inspect_event: Separate paged fields with \t.
- cli: inspect: Refactor command.
- cli: cli: Add option shortcuts.
- cli: inspect: Make two subcommands for events and publications.
- cli: cli: Add some consistency to help messages.
- installed aerich
- add aerich upgrade before init
- add aerich configuration
- add initial configuration, replace get_configuration with toml.load to avoid validation error on aerich init
- add aenrich with migration initialized and auto-upgrade on start
- re-init
- added new test
- changed name of __implement_db_changes__ to _implement_db_changes and using logging facility for printing starting update database
- added log
- prova.txt
- improved escape characters for telegram
- move aerich conf to storage.db
- remove test file
- change first script name
- handle aerich migrations not found
- ignore test_run.sh
- ignore pylintrc
- completed chain, handling exception
- cli: Move settings option to subcommands.
- docker: mobilizon-reshare: Upstream dynaconf. (#115)
- fixed cli
- fixed exit on migration exception
- Revert "ignore pylintrc"
- add intermediate variable since aerich doesn't play nice with functions as configuration entrypoint
- Update guix package definition after core-updates merge. (#122)
- retry command (#119)
- Run CI tests for every push to master (#123)
- Rewrite README.md. (#125)
- rework config (#121)
- docker: Upstream python-markdownify. (#126)
- Add Docker Hub link to README.md. (#127)
- Fix markdown cropping (#128)
- added timezone_sensitive marker (#129)
- moved run tests to script
- Use test script everywhere in the pipeline. (#130)
- Debug image (#131)
- added scheduling (#133)
- Python scheduler docker image (#134)
- Add event name to notifications. (#136)
- Update README.md
- Feature/optimizing download (#118)
- added default (#138)
- rework maintenance cli (#140)
- Release v0.2.0.
- Enforce the same dependencies in dev and prod. (#141)
- fixed facebook format (#142)
- fixed missing whitespace (#143)
- scheduler.py: Fix intervals. (#146)
- README.md: Add Pypi link. (#148)
- fix timezone in test (#145)
- Pave the way for using only EventPublications outside of storage module. (#149)
- Feature/autogenerate api doc (#147)
- Release v0.2.1.
- mobilizon-reshare.scm: Hotfix applied to the image
- Release v0.2.2.
- Restore tests in Guix package.
- platform tests (#150)
- test commands (#151)
- set timezone to Rome (#153)
- set timezone to CUT
- fix event retry (#152)
- Release v0.2.3.
- Upstream python-tweepy. (#155)
- Run CI for pull requests. (#159)
- Update twitter.tmpl.j2 (#160)
- Pull publish (#157)
- Fix recap (#161)
- Update docker-compose.yml
- Release v0.3.0.
- Add completion scripts. (#166)
- introduce locale (#165)
- Hotfix release script.
- Release v0.3.1.
- Add publish command. (#167)
- Move Guix packages to a channel. (#168)
- Release v0.3.2.
- add dry run (#169)
- add dry run (#172)
- Update dependencies. (#170)
- Update development-environment-with-guix.md
- refactor coordinators (#171)
- added poc
- added check for sqlite db
- added events test
- draft docker-compose-test.yml
- improved docker-compose
- added support for postgres migrations
- add documentation
- added some qol to migrations
- added migration generation script
- removed settings.toml
- waiting for postgress in script
- commented script
